pinot-core/src/main/java/com/linkedin/pinot/core/data/recordtransformer/CompoundTransformer.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (C) 2014-2018 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.data.recordtransformer;
+
+import com.linkedin.pinot.common.data.Schema;
+import com.linkedin.pinot.core.data.GenericRow;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.Nullable;
+
+
+/**
+ * The {@code CompoundTransformer} class performs multiple transforms based on the inner {@link RecordTransformer}s.
+ */
+public class CompoundTransformer implements RecordTransformer {
+  private final List<RecordTransformer> _transformers;
+
+  /**
+   * Returns a record transformer that performs time transform, expressions transform and data type transform.
+   */
+  public static CompoundTransformer getDefaultTransformer(Schema schema) {
+    return new CompoundTransformer(
+        Arrays.asList(new TimeTransformer(schema), new ExpressionTransformer(schema), new DataTypeTransformer(schema)));
```

Comment:
Can you add a comment (preferably a story) here about how certain transformers have to be before others? The CompoundTransformer class somehow implies that the order does not matter, but it does, doesn't it?